inst/include/model_output.hpp

```diff
@@ -10,68 +10,85 @@
+  std::copy_n(state.base.p_infections.data(), state.base.p_infections.size(), REAL(r_p_infections));
+  std::copy_n(state.base.h_hiv_deaths_art.data(), state.base.h_hiv_deaths_art.size(), REAL(r_h_hiv_deaths_art));
+  std::copy_n(state.base.h_art_initiation.data(), state.base.h_art_initiation.size(), REAL(r_h_art_initiation));
+  std::copy_n(state.base.p_hiv_deaths.data(), state.base.p_hiv_deaths.size(), REAL(r_p_hiv_deaths));
+
+  Rcpp::List ret(12);
+  Rcpp::CharacterVector names(12);
+  names[0] = "p_total_pop";
+  ret[0] = r_p_total_pop;
+  names[1] = "births";
+  ret[1] = r_births;
+  names[2] = "p_total_pop_natural_deaths";
+  ret[2] = r_p_total_pop_natural_deaths;
+  names[3] = "p_hiv_pop";
+  ret[3] = r_p_hiv_pop;
+  names[4] = "p_hiv_pop_natural_deaths";
+  ret[4] = r_p_hiv_pop_natural_deaths;
+  names[5] = "h_hiv_adult";
+  ret[5] = r_h_hiv_adult;
+  names[6] = "h_art_adult";
+  ret[6] = r_h_art_adult;
+  names[7] = "h_hiv_deaths_no_art";
+  ret[7] = r_h_hiv_deaths_no_art;
+  names[8] = "p_infections";
+  ret[8] = r_p_infections;
+  names[9] = "h_hiv_deaths_art";
+  ret[9] = r_h_hiv_deaths_art;
+  names[10] = "h_art_initiation";
+  ret[10] = r_h_art_initiation;
+  names[11] = "p_hiv_deaths";
+  ret[11] = r_p_hiv_deaths;
+  ret.attr("names") = names;
 
-  std::copy_n(state.p_total_pop.data(), state.p_total_pop.size(), REAL(r_p_total_pop));
-  std::copy_n(state.births.data(), state.births.size(), REAL(r_births));
-  std::copy_n(state.p_total_pop_natural_deaths.data(), state.p_total_pop_natural_deaths.size(), REAL(r_p_total_pop_natural_deaths));
-  std::copy_n(state.p_hiv_pop.data(), state.p_hiv_pop.size(), REAL(r_p_hiv_pop));
-  std::copy_n(state.p_hiv_pop_natural_deaths.data(), state.p_hiv_pop_natural_deaths.size(), REAL(r_p_hiv_pop_natural_deaths));
-  std::copy_n(state.h_hiv_adult.data(), state.h_hiv_adult.size(), REAL(r_h_hiv_adult));
-  std::copy_n(state.h_art_adult.data(), state.h_art_adult.size(), REAL(r_h_art_adult));
-  std::copy_n(state.h_hiv_deaths_no_art.data(), state.h_hiv_deaths_no_art.size(), REAL(r_h_hiv_deaths_no_art));
-  std::copy_n(state.p_infections.data(), state.p_infections.size(), REAL(r_p_infections));
-  std::copy_n(state.h_hiv_deaths_art.data(), state.h_hiv_deaths_art.size(), REAL(r_h_hiv_deaths_art));
-  std::copy_n(state.h_art_initiation.data(), state.h_art_initiation.size(), REAL(r_h_art_initiation));
-  std::copy_n(state.p_hiv_deaths.data(), state.p_hiv_deaths.size(), REAL(r_p_hiv_deaths));
+  if constexpr (ModelVariant::run_child_model) {
+    constexpr auto base = ss.base;
+    constexpr auto children = ss.children;
+    Rcpp::NumericVector r_hc_hiv_pop(base.hDS * children.hTM * base.pAG * base.NS * output_years);
+    r_hc_hiv_pop.attr("dim") = Rcpp::NumericVector::create(base.hDS, children.hTM, base.pAG, base.NS, output_years);
+    std::copy_n(state.children.hc_hiv_pop.data(), state.children.hc_hiv_pop.size(), REAL(r_hc_hiv_pop));
+    ret.push_back(r_hc_hiv_pop, "hc_hiv_pop");
+  }
```

Comment:
Should we return here a list of lists? `base`, `child` etc. and should the input data also be split like this?

Reply:
Will address this in another PR immediately after this one

Reply:
Actually, made a ticket for this https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-4516/Split-the-input-and-output-data-into-lists-of-lists-split-by-model-type